FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -125,18 +130,29 @@ final class MessageService {
                 isEmail: true
             )
 
-            let processedMessage = self.processMessage(rawMimeData: rawMimeData, with: decrypted)
+            let processedMessage = try self.processMessage(rawMimeData: rawMimeData, with: decrypted, keys: keys)
             resolve(processedMessage)
         }
     }
 
-    private func processMessage(rawMimeData: Data, with decrypted: CoreRes.ParseDecryptMsg) -> ProcessedMessage {
+    private func processMessage(rawMimeData: Data, with decrypted: CoreRes.ParseDecryptMsg, keys: [PrvKeyInfo]) throws -> ProcessedMessage {
         let decryptErrBlocks = decrypted.blocks
             .filter { $0.decryptErr != nil }
 
-        let attachments = decrypted.blocks
+        let attachments = try decrypted.blocks
             .filter(\.isAttachmentBlock)
-            .map(MessageAttachment.init)
+            .map { block -> MessageAttachment in
+                var name = block.attMeta?.name ?? "Attachment"
+                let size = block.attMeta?.length ?? 0
+                var data = block.attMeta?.data ?? Data()
+
+                if name.fileExtension == Constants.encryptedAttachmentExtension {
+                    data = (try core.decryptFile(encrypted: data, keys: keys, msgPwd: nil).content)
+                    name = name.dropExtension()
+                }
```

Comment:
Just realized, please look closely at the BlockType. If it's returned as `encryptedFile` then you decrypt it, else leave as is. This means you don't have to check the file name at all.

https://github.com/FlowCrypt/flowcrypt-ios/blob/ac29a9c92908d0d71c67c53b099080ecfcae5837/FlowCrypt/Core/CoreTypes.swift#L200

Reply:
done